website/views/organization.py

```diff
@@ -1590,11 +1592,123 @@ def feed(request):
     # Determine if pagination is required
     is_paginated = page_obj.has_other_pages()
 
+    # Check if the user has the mentor badge
+    if request.user.is_authenticated:
+        is_mentor = UserBadge.objects.filter(user=request.user, badge__title="Mentor").exists()
+    else:
+        is_mentor = False
+
     return render(
         request,
         "feed.html",
         {
             "page_obj": page_obj,
             "is_paginated": is_paginated,  # Pass this flag to the template
+            "is_mentor": is_mentor,  # Add is_mentor to the context
         },
     )
+
+
+@login_required
+@require_POST
+def like_activity(request, id):
+    activity = get_object_or_404(Activity, id=id)
```

Comment:
The like and dislike functionality for issues differs significantly from the implementation for activities:

Issue likes and dislikes are linked directly to the user profile model (issue_upvoted, issue_downvoted).
However, we don’t want to store activity likes and dislikes in a similar manner, as it would be redundant.

Currently, an email is sent to the user every time their issue is liked. Sending another email when their issue is liked via the activity feed would be redundant.
That said, we can send an email notification if their content is featured on Blue Sky.

Conclusion:
While this can be implemented, it would require many if-else conditions, making the code more complex.